developer/services/qrexec3.md

```diff
@@ -180,173 +150,136 @@ By contrast, consider this command:
+This means it is also possible to install a different script for a particular service argument.
 
 See below for an example service using an argument.
 
-### Revoking "Yes to All" authorization ###
+### Revoking "Yes to All" authorization
 
-Qubes RPC policy supports "ask" action. This will prompt the user whether given
-RPC call should be allowed. That prompt window also has a "Yes to All" option,
-which will allow the action and add a new entry to the policy file, which will
-unconditionally allow further calls for the given service-srcVM-dstVM tuple.
+Qubes RPC policy supports "ask" action.
+This will prompt the user whether given RPC call should be allowed.
+That prompt window also has a "Yes to All" option, which will allow the action and add a new entry to the policy file, which will unconditionally allow further calls for the given service-srcVM-dstVM tuple.
 
-In order to remove such authorization, issue this command from a dom0 terminal
-(for `qubes.Filecopy` service):
+In order to remove such authorization, issue this command from a dom0 terminal (for `qubes.Filecopy` service):
 
     sudo nano /etc/qubes-rpc/policy/qubes.Filecopy
 
-and then remove the first line(s) (before the first `##` comment) which are
-the "Yes to All" results.
-
+and then remove the first line(s) (before the first `##` comment) which are the "Yes to All" results.
 
-### Qubes RPC example ###
+### Qubes RPC example
 
-We will show the necessary files to create an rpc call that adds two integers
-on the target and returns back the result to the invoker.
+As a demonstration, we can create an RPC service that adds two integers in a target domain (the server, call it "anotherVM") and returns back the result to the invoker (the client, "someVM").
+In someVM, create a file with the following contents and save it with the path `/usr/bin/our_test_add_client`:
 
- * rpc client code (`/usr/bin/our_test_add_client`):
+    #!/bin/sh
+    echo $1 $2             # pass data to RPC server
+    exec cat >&$SAVED_FD_1 # print result to the original stdout, not to the other RPC endpoint
 
-       #!/bin/sh
-       echo $1 $2    # pass data to rpc server
-       exec cat >&$SAVED_FD_1 # print result to the original stdout, not to the other rpc endpoint
-
- * rpc server code (*/usr/bin/our\_test\_add\_server*)
+Our server will be anotherVM at `/usr/bin/our_test_add_server`.
+The code for this file is:
 
-       #!/bin/sh
-       read arg1 arg2 # read from stdin, which is received from the rpc client
-       echo $(($arg1+$arg2)) # print to stdout - so, pass to the rpc client
+    #!/bin/sh
+    read arg1 arg2        # read from stdin, which is received from the RPC client
+    echo $(($arg1+$arg2)) # print to stdout, which is passed to the RPC client
 
- * policy file in dom0 (*/etc/qubes-rpc/policy/test.Add* )
+We'll need to create a service called `test.Add` with its own definition and policy file in dom0.
```

Comment:
Hmm, maybe better first create `/etc/qubes-rpc/test.Add` in `anotherVM` and only later the policy? Then actions in `anotherVM` would be one after another.